### PR TITLE
[PWGLF] Change V0 Cos PA to regular column for filterability

### PR DESCRIPTION
**Pull request by ddobrigk**
This (probably long overdue) change in the strangeness data model moves the V0 Cos PA to a regular column so that declarative filtering (much faster) will work on this variable. Together with the previous changes to the strangeness data model (#3829), even parametric CosPA cuts will be applicable as p and pT are now expression columns instead of dynamic columns. Note that the change is, among other reasons, also justified because the collision received as argument is never changed with respect to the collision that the V0 is associated to, as far as I know. If there is any use case for that, please tell me. 

Note that a lot of files are touched by this commit. I tried to be as careful as possible, but if anyone spots anything that looks suspicious, please let me know and I promise to react as quickly as I manage. Thank you!!

### CODE REVIEW

**Review by aalkin — APPROVED** [review]

**Comment by ddobrigk** [comment]
One important consideration for HF (@fgrosa @fcolamar @adubla): there is a change in the HF candidate v0cospa column too, and now that I think of it, I hope it does not create troubles with existing derived data...

**Review by njacazio — DISMISSED** [review]

**Review by fgrosa — DISMISSED** [review]

**Comment by fgrosa** [comment]
> One important consideration for HF (@fgrosa @fcolamar @adubla): there is a change in the HF candidate v0cospa column too, and now that I think of it, I hope it does not create troubles with existing derived data...

Hi @ddobrigk there is no issue for our derived data, because they only contain track (and in the future V0 and cascade) indices, so green lights from HF side!

**Comment by dsekihat** [comment]
Dear David, it looks OK from DQ side.

**Review by dsekihat — APPROVED** [review]

**Comment by ddobrigk** [comment]
Thanks a lot, @fgrosa @dsekihat @njacazio and @aalkin ! Could the others also please take a look when they have a moment? Thanks for your patience! :-)

**Review by mfaggin — DISMISSED** [review]

**Comment by ddobrigk** [comment]
Thank you, @mfaggin!

**Review by saganatt — DISMISSED** [review]

**Comment by ddobrigk** [comment]
Thanks @saganatt !

**Review by iarsene — DISMISSED** [review]

**Review by victor-gonzalez — APPROVED** [review]
Thanks!!

File: PWGEM/PhotonMeson/TableProducer/createPCM.cxx [hunk 1]
```diff
@@ -299,7 +299,8 @@ struct createPCM {
              svpos[0], svpos[1], svpos[2],
              pvec0[0], pvec0[1], pvec0[2],
              pvec1[0], pvec1[1], pvec1[2],
-             v0dca, pos.dcaXY(), ele.dcaXY());
+             v0dca, pos.dcaXY(), ele.dcaXY(),
+             v0CosinePA, v0dca);
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/TableProducer/createPCM.cxx:303]: Looks like the v0dca doubled here. I guess the last one can be removed?

Comment (ddobrigk) [reply PWGEM/PhotonMeson/TableProducer/createPCM.cxx:303]: Actually, no, this is a bug, sorry. The v0dca in the code is the DCA between V0 daughters and the last item in the V0Data table should be the DCA of the V0 to the PV ... Let me fix that. Thanks a lot for looking!!!

**Review by mhemmer-cern — APPROVED** [review]

**Review by saganatt — DISMISSED** [review]

**Review by njacazio — APPROVED** [review]

**Review by mhemmer-cern — APPROVED** [review]

**Review by aalkin — APPROVED** [review]

**Review by victor-gonzalez — APPROVED** [review]

**Review by iarsene — APPROVED** [review]

**Review by fgrosa — APPROVED** [review]

**Review by njacazio — APPROVED** [review]

**Review by lauraser — APPROVED** [review]

**Review by aalkin — APPROVED** [review]

**Review by ercolessi — APPROVED** [review]

**Pull request merged**